spring-test/src/main/java/org/springframework/mock/web/MockHttpServletRequest.java

```diff
@@ -410,13 +410,21 @@ public void setCharacterEncoding(@Nullable String characterEncoding) {
 	private void updateContentTypeHeader() {
 		if (StringUtils.hasLength(this.contentType)) {
 			String value = this.contentType;
+			if (isJsonContentType(value) {
+				return;	
+			}
+			
 			if (StringUtils.hasLength(this.characterEncoding) && !this.contentType.toLowerCase().contains(CHARSET_PREFIX)) {
 				value += ';' + CHARSET_PREFIX + this.characterEncoding;
 			}
 			doAddHeaderValue(HttpHeaders.CONTENT_TYPE, value, true);
 		}
 	}
 
+	private boolean isJsonContentType(String contentType) {
+		return contentType.toLowerCase().startsWith("application/json");
```

Comment:
Would be nice to validate that the contentType is not null :)

An approach would be:
return contentType != null && contentType.toLowerCase().startsWith("application/json");